core/templates/dev/head/domain/skill/SkillUpdateService.ts

```diff
@@ -259,6 +269,24 @@ angular.module('oppia').factory('SkillUpdateService', [
               misconception.setFeedback(oldFeedback);
             });
         }
+      },
+
+      updateRubricForDifficulty: function(skill, difficulty, explanation) {
+        if (SKILL_DIFFICULTIES.indexOf(difficulty) === -1) {
+          throw Error('Invalid difficulty value passed');
+        }
+        var oldExplanation = skill.getRubricExplanation(difficulty);
+        _applyRubricPropertyChange(
+          skill, difficulty, explanation,
+          function(changeDict, skill) {
+            skill.updateRubricForDifficulty(difficulty, explanation);
+          }, function(changeDict, skill) {
+            if (oldExplanation !== null) {
+              skill.updateRubricForDifficulty(difficulty, oldExplanation);
+            } else {
+              skill.removeRubric(difficulty);
```

Comment:
Do you actually want to support removeRubric functionality?

One thing to perhaps consider is to do this change as a "blob" -- i.e. replace old "rubrics" with new "rubrics". I think that would be fine, and potentially simpler. Up to you though.

Reply:
Removed the function. It was only needed to revert an add rubric function, but since we are having 3 rubrics for skills from skill creation itself, that would never happen.